contract/src/msgs/data_requests/sudo/remove_requests.rs

```diff
@@ -130,21 +137,32 @@ fn remove_request_and_process_distributions(
     state::remove_request(deps.storage, dr_id)?;
     DR_ESCROW.remove(deps.storage, &dr_id);
 
-    Ok((event, bank_messages))
+    Ok((event, bank_messages, stakers_effected))
 }
 
 impl SudoHandler for remove_requests::Sudo {
     fn sudo(self, mut deps: DepsMut, _: Env) -> Result<Response, ContractError> {
         let token = TOKEN.load(deps.storage)?;
         let mut response = Response::new();
+
+        let mut all_stakers_effected = HashSet::new();
         for removal in self
             .requests
             .into_iter()
             .map(|(dr_id, messages)| remove_request_and_process_distributions(dr_id, &messages, &mut deps, &token))
         {
-            let (event, bank_messages) = removal?;
+            let (event, bank_messages, stakers_effected) = removal?;
+            all_stakers_effected.extend(stakers_effected);
             response = response.add_event(event).add_messages(bank_messages);
         }
+
+        for staker in all_stakers_effected {
```

Comment:
Is the order of a hashset deterministic? I imagine CosmWasm would have thrown a fit if it isn’t, but just wanted to make sure 😬

Reply:
No rust HashSet is not deterministic. Does it matter?